store/store.go

```diff
@@ -1109,9 +1110,13 @@ func (s *Store) Download(name string, downloadInfo *snap.DownloadInfo, pbar prog
 		url = downloadInfo.DownloadURL
 	}
 
-	if err := download(name, url, user, s, w, pbar); err != nil {
+	hash, err := download(name, url, user, s, w, pbar)
+	if err != nil {
 		return "", err
 	}
+	if downloadInfo.Sha3_384 != "" && hash != downloadInfo.Sha3_384 {
+		return "", fmt.Errorf("hashsum mismatch for %s: got %s but expected %s", w.Name(), hash, downloadInfo.Sha3_384)
```

Comment:
```
(`"sha3-384 mismatch downloading %q: got %s but expected %s"`, name, hash, downloadInfo.Sha3_384)
```
I suggest using the snap name there instead of the temporary filename.